modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java

```diff
@@ -52,7 +51,8 @@ public ElasticsearchContainer(String dockerImageName) {
         addExposedPorts(ELASTICSEARCH_DEFAULT_PORT, ELASTICSEARCH_DEFAULT_TCP_PORT);
         setWaitStrategy(new HttpWaitStrategy()
             .forPort(ELASTICSEARCH_DEFAULT_PORT)
-            .forStatusCodeMatching(response -> response == HTTP_OK || response == HTTP_UNAUTHORIZED)
+            .forPath("/_cluster/health")
```

Comment:
This endpoint is probably not available either when the cluster is secured AFAIK. I don't think this can work as is IMO.
I'd let the default behavior.

But we can add this nice idea in #2321 which adds a simple `withPassword()` method which makes the cluster secured. There we can add 

```
.withBasicCredentials("elastic", password)
```

and just check the `HTTP_OK` status.

Reply:
Yes, you are correct. This is what I have described in the comment as well: https://github.com/testcontainers/testcontainers-java/issues/2982#issuecomment-658401897

(Sorry for causing confusion by keeping most of the conversation on the issue, not the pull request)